packages/styled/types/tests.tsx

```diff
@@ -51,3 +51,17 @@ const Container2 = styled.div<{ test: number }>(props => ({
 const Container3 = styled.div(({ theme }) => ({
   width: theme.width
 }))
+
+// This example shows using shouldForwardProps to define a prop
+// who's type clashes with an intrinsic prop.
+// It makes use of a custom type guard on shouldForwardProp to exclude color
+
+export const Box = styled('div', {
+  shouldForwardProp: (
+    propName
+  ): propName is Exclude<keyof JSX.IntrinsicElements['div'], 'color'> =>
```

Comment:
this is somewhat an example of how TS is unsound 😅 Not a big deal here though - it's just an example

Reply:
Yeah, it would be great if TypeScript could generate a type guard signature based on the type narrowing inside a function like this. May be worth raising an issue?